### ISSUE TITLE
Errors with symlinks and Google Drive

### ISSUE THREAD

**Issue by parasiteoflife** [rclone/rclone#4581]
#### What is the problem you are having with rclone?

Rclone is showing errors with symlinks. I've tried three times and I still get the same errors. Every single symlink shows that error (yes, the path has 634 symbolic links). If it matters the path points to a mounted veracrypt container formated as NTFS.

#### What is your rclone version (output from `rclone version`)

1.53.0

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Windows 10 x64

####  Which cloud storage system are you using? (eg Google Drive)

Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

`rclone copy Y:\ Crypt:Boo --verbose --no-update-modtime --transfers 14 --checkers 16 --contimeout 60s --timeout 300s --retries 3 --low-level-retries 10 --exclude /$RECYCLE.BIN/** --exclude /System Volume Information/** --create-empty-src-dirs --backup-dir Crypt:Rem --links --bwlimit Mon-08:00,7M 23:59,off Tue-08:00,7M 23:59,off --max-size 30M --stats 1s --stats-file-name-length 0 --fast-list`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

I see a lot of lines like this and also this is a piece of the end of the log:

```
2020/09/10 01:59:31 ERROR : redacted.mkv.rclonelink: corrupted on transfer: sizes differ 0 vs 106
2020/09/10 01:59:31 INFO  : redacted.rclonelink: Removing failed copy
2020/09/10 01:59:31 ERROR : redacted2.mp4.rclonelink: corrupted on transfer: sizes differ 0 vs 37
2020/09/10 01:59:31 INFO  : redacted2.mp4.rclonelink: Removing failed copy
2020/09/10 01:59:31 ERROR : redacted3.mkv.rclonelink: corrupted on transfer: sizes differ 0 vs 87
2020/09/10 01:59:31 INFO  : redacted3.mkv.rclonelink: Removing failed copy
2020/09/10 01:59:31 INFO  :
Transferred:   	  218.229k / 217.749 kBytes, 100%, 357 Bytes/s, ETA -
Errors:               634 (retrying may help)
Checks:             68649 / 68649, 100%
Transferred:            0 / 4, 0%
Elapsed time:     20m25.7s
Transferring:
* redacted4.rclonelink:  0% /0, 19/s, -
* redacted3.mp4.rclonelink:  0% /0, 16/s, -
* redacted2.mkv.rclonelink:  0% /0, 26/s, -
* redacted.mkv.rclonelink:  0% /0, 30/s, -

2020/09/10 01:59:32 INFO  : There was nothing to transfer
2020/09/10 01:59:32 ERROR : Attempt 3/3 failed with 634 errors and: corrupted on transfer: sizes differ 0 vs 87
2020/09/10 01:59:32 INFO  :
Transferred:   	  218.229k / 218.229 kBytes, 100%, 356 Bytes/s, ETA 0s
Errors:               634 (retrying may help)
Checks:             68649 / 68649, 100%
Elapsed time:     20m26.7s

2020/09/10 01:59:32 Failed to copy with 634 errors: last error was: corrupted on transfer: sizes differ 0 vs 87
```

**Comment by ncw** [comment]
Can you try to make a simple reproducer for this - something I can try here?

Thanks

**Comment by parasiteoflife** [comment]
Hey, I don't know what to tell you exactly. I have just tested it doing this:

> - Copy some files as symlinks.
> - Sync the folder to a crypt remote (pointing to a Google Drive remote).
> - Rclone will show errors on every symlink.

Btw, the crypt remote has standard and folder name encryption.  


Maybe this has to do with the fact that rclone stores the symlinks as a text with the source path thus it occupies more than 0 bytes on the remote against the 0 bytes the symlinks occupy on the local storage?

**Comment by ncw** [comment]
This doesn't reproduce on linux so I tested this on Windows too.

I'm not a Windows power user so I'll describe what I did

I made a JUNCTION with

```
Z:\go\src\github.com\rclone\rclone>mklink /j c:\a\file-link.txt c:\a\file.txt
Junction created for c:\a\file-link.txt <<===>> c:\a\file.txt
```

Then listed the directory with rclone - I can see here that the length of the junction is wrong - it should be the length of the destination of the symlink

```
Z:\go\src\github.com\rclone\rclone>rclone lsl --links c:\a
        0 2020-09-25 18:13:42.422387300 file-link.txt.rclonelink
     6042 2020-09-18 15:34:20.678820900 file.txt
```

This does indeed create the error you described

```
Z:\go\src\github.com\rclone\rclone>rclone copy --links -vv c:\a drive:test/a --retries 1
2020/09/25 18:20:02 DEBUG : rclone: Version "v1.54.0-DEV" starting with parameters ["rclone" "copy" "--links" "-vv" "c:\\a" "drive:test/a" "--retries" "1"]
[snip]
2020/09/25 18:20:04 ERROR : file-link.txt.rclonelink: corrupted on transfer: sizes differ 0 vs 13
2020/09/25 18:20:04 INFO  : file-link.txt.rclonelink: Removing failed copy
```

I'm a bit confused about windows links. Which type of link did you use - was it a junction point?

I've attempted to fix this - can you give this a go please?

[v1.54.0-beta.4785.6e15cf51c.fix-4581-win-links](https://beta.rclone.org/branch/fix-4581-win-links/v1.54.0-beta.4785.6e15cf51c.fix-4581-win-links/) on branch [fix-4581-win-links](https://github.com/rclone/rclone/tree/fix-4581-win-links) (uploaded in 15-30 mins)

**Comment by parasiteoflife** [comment]
I use symbolic links which is the default for mklink (for files), I don't think I've used junctions on Windows.  
I tried to download a Windows build of that branch but at the time of writing there is no Win build available, can you please check that?

**Comment by ncw** [comment]
Thanks for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.54
